src/main/kotlin/lotto/domain/PurchasedLottoTickets.kt

```diff
@@ -0,0 +1,21 @@
+package lotto.domain
+
+class PurchasedLottoTickets(val purchasedCount: Int, lottoNumberGenerator: () -> Set<Int>) {
+    val purchasedLottoTickets: List<LottoTicket> = generateLottoTickets(lottoNumberGenerator)
+
+    init {
+        require(purchasedCount >= PURCHASED_COUNT_MIN_VALUE) { INVALID_PURCHASED_COUNT_MESSAGE }
+    }
+
+    private fun generateLottoTickets(lottoNumberGenerator: () -> Set<Int>): List<LottoTicket> {
+        val lottoTickets: MutableList<LottoTicket> = mutableListOf()
+        repeat(purchasedCount) { lottoTickets.add(LottoTicket(lottoNumberGenerator)) }
```

Comment:
`List()` 생성자를 사용하면 아래와 같이도 작성이 가능해요!! 
 다만 `purchasedCount`가 음수일때의 동작방식이 repeat와 달라 현재 테스트가 실패할 수 있겠네요!

`lottoTickets` 지역변수가 정말로 필요한지 고민해보고, 이에따라 코드를 수정해보시면 좋을 것 같습니다!

```suggestion
        return List(purchasedCount) { LottoTicket(lottoNumberGenerator) }
```

Reply:
😮 와.. 이러한 형태의 `List()` 생성자가 가능한지 모르고 있었는데, 덕분에 좋은 꿀팁을 얻게 되었습니다!

같은 결과를 갖기 위한 로직에서 제안해주신 `List()` 생성자를 활용하는 코드가 더 이해하기 쉬운 직관적인 코드라고 느껴져

굳이 `lottoTickets` 지역변수를 활용하지 않아도 될 것 같습니다ㅎㅎ